README.md
<img src="http://geo.azmag.gov/maps/wickenburg/img/townLogo_black.jpg" alt="wickenburg" height="80"/>  ![](http://geo.azmag.gov/maps/readonaz/app/resources/img/maglogo_black.png)

# Town of Wickenburg Zoning Viewer #
MAG mapping application for the town of Wickenburg showing zoning data

[![Website](https://img.shields.io/website-up-down-green-red/http/shields.io.svg?label=my-website)](http://geo.azmag.gov/maps/wickenburg/)
[![GitHub version](https://badge.fury.io/gh/AZMAG%2Fmap-Wickenburg.svg)](https://badge.fury.io/gh/AZMAG%2Fmap-Wickenburg)
[![GitHub issues](https://img.shields.io/github/issues/AZMAG/map-Wickenburg.svg)](https://github.com/AZMAG/map-Wickenburg/issues)
[![dependencies](https://david-dm.org/AZMAG/map-Wickenburg.png)](https://david-dm.org/AZMAG/map-Wickenburg)
[![devDependency Status](https://david-dm.org/AZMAG/map-Wickenburg/dev-status.png)](https://david-dm.org/AZMAG/map-Wickenburg)
[![Semver](http://img.shields.io/SemVer/2.0.0.png)](http://semver.org/spec/v2.0.0.html)
[![Built with Grunt](http://cdn.gruntjs.com/builtwith.png)](http://gruntjs.com/)
[![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](https://opensource.org/licenses/MIT)

****************************************************************************************

This JavaScript web app viewer was developed by MAG staff for the Town of Wickenburg using ESRI's JavaScript API.  It is used to view town zoning, localized flood areas, and parcel boundaries for both Maricopa County and Yavapai County.

## Keywords

`Zoning` `Flood Areas` `Parcels` `Employers` `Wickenburg` `Town` `Maricopa County` `Yavapai County` `Maps` `MAG` `ESRI` `#MAGmaps`

## Version

### version 3.9.1 ###

 `Updated: 2020-02-14`

 `Created: 2013-04-01`

Releases will be numbered with the following format:

**`<major>.<minor>.<patch>`**

And constructed with the following guidelines:

1. **MAJOR** version when you make incompatible API changes **bumps the major** resets minor and patch
2. **MINOR** version when you add functionality in a backwards-compatible manner **bumps the minor** resets patch
3. **PATCH** version when you make backwards-compatible bug fixes and misc changes **bumps only the patch**

## Credits

* Maricopa Association of Governments (MAG) and the MAG member agencies

## Site URL

* [http://geo.azmag.gov/maps/wickenburg/](http://geo.azmag.gov/maps/wickenburg/)

## Documentation

A detailed list of the changes and fixes in each version update can be found in the `CHANGELOG.md`.

* [CHANGELOG](CHANGELOG.md)

## Resources

## Dependencies

* [git-labelmaker](https://github.com/himynameisdave/git-labelmaker?utm_source=hashnode.com)

## Legal Disclaimer

The Town of Wickenburg and Maricopa Association of Governments provides the data within these pages as a public resource of general information for use "as is." The Town of Wickenburg and Maricopa Association of Governments GIS (Geographic Information System) departments provides this information with the understanding that it is not guaranteed to be accurate, correct or complete and any conclusions drawn from such information are the sole responsibility of the user. Further, the Town of Wickenburg and Maricopa Association of Governments GIS departments makes no warranty, representation or guaranty as to the content, sequence, accuracy, timeliness or completeness of any of the spatial or database information provided herein. While every effort has been made to ensure the content, sequence, accuracy, timeliness or completeness of materials presented within these pages, the Town of Wickenburg and Maricopa Association of Governments GIS Departments assumes no responsibility for errors or omissions, and explicitly disclaims any representations and warranties, including, without limitation, the implied warranties of merchantability and fitness for a particular purpose. The Town of Wickenburg and Maricopa Association of Governments GIS Departments shall assume no liability for:

    1. Any errors, omissions, or inaccuracies in the information provided, regardless of how caused; or
    2. Any decision made or action taken or not taken by viewer in reliance upon any information or data furnished hereunder.

 Availability of the Town of Wickenburg or Maricopa Association of Governments Map Server is not guaranteed. Applications, servers, and network connections may be unavailable at any time for maintenance or unscheduled outages. Outages may be of long duration. Users are cautioned to create dependencies on these services for critical needs.

THE FOREGOING WARRANTY IS EXCLUSIVE AND IN LIEU OF ALL OTHER WARRANTIES OF MERCHANTABILITY, FITNESS FOR PARTICULAR PURPOSE AND/OR ANY OTHER TYPE WHETHER EXPRESSED OR IMPLIED. In no event shall The Town of Wickenburg and Maricopa Association of Governments become liable to users of these data, or any other party, for any loss or direct, indirect, special, incidental or consequential damages, including, but not limited to, time, money or goodwill, arising from the use or modification of the data.

 To assist The Town of Wickenburg and Maricopa Association of Governments in the maintenance and/or correction of the data, users should provide the Town of Wickenburg and Maricopa Association of Governments GIS Departments with information concerning errors or discrepancies found in using the data. Please use the e-mail contact address at the bottom of the affected web page.

 Please acknowledge the Town of Wickenburg GIS as the source when Map Server data is used in the preparation of reports, papers, publications, maps, or other products.

## Copyright and Licensing

Code released under the MIT license.

A copy of the license is available in the repository's `LICENSE` file.

* [LICENSE](LICENSE)


CHANGELOG.md
# CHANGELOG #
### Wickenburg Zoning Website ###

[![Semver](http://img.shields.io/SemVer/2.0.0.png)](http://semver.org/spec/v2.0.0.html)

All notable changes to this project will be documented in this file.

Releases will be numbered with the following format:

**`<major>.<minor>.<patch>`**

And constructed with the following guidelines:

1. **MAJOR** version when you make incompatible API changes **bumps the major** resets minor and patch
2. **MINOR** version when you add functionality in a backwards-compatible manner **bumps the minor** resets patch
3. **PATCH** version when you make backwards-compatible bug fixes and misc changes **bumps only the patch**

#### Types of Changes ####

* `Added` for new features
* `Changed` for changes in existing functionality
* `Deprecated` for soon-to-be removed features
* `Removed` for now removed features
* `Fixed` for any bug fixes
* `Security` in case of vulnerabilities

[Source](http://keepachangelog.com/en/1.0.0/)

***********************************************

## Outstanding Items

* zoom issue with geocoder
* turn off measurement tool when window closes

***********************************************

## Version 3.9.1 | 2020-02-14

* Added Opportunity Zones to map

## Version 3.9.0 | 2019-11-06

* Changed field names in Parcel Layer (PARCEL_APN_LABEL to PARCEL, P_ADDRESS to ADDRESS)

## Version 3.8.1 | 2019-09-11

* Fixed service URL
* Fixed logo urls
* Added toggle npm to help build process

## Version 3.8.0 | 2017-06-29

* updates to data
* updates to package.json
* updates to Gruntfiles
* add dist/ folder for build
* updated [ArcGIS API for JavaScript - v3.20](https://developers.arcgis.com/javascript/3/)
* updates to about menu dates mostly

## Version 3.7.8 | 2017-02-14

* updated link to employment viewer pdf data
* updated link to demographic viewer pdf data

## Version 3.7.7 | 2016-09-16

* updates to Town Boundary
* updates to Zoning
* take zoning layer off of initial loading

## Version 3.7.6 | 2016-06-22

* fix URL for Maricopa County Assessors Office

## Version 3.7.5 | 2016-02-26

* added basemap transparency slider
* disabled basemaptoggle button
* updated to latest arcgis javascript api version
* upgraded "geocoder" widget to newer "search" widget
* updated the way layers are added and simplified popup handling
* added new layers (BLI and Subdivisions)
* added markup tools
* make popup title smaller or something so it doesn't run into the buttons
* subdivisions layer remove acres/sq mi
* add info link to parcel
* reorder legend layers to have zoning at the top
* fix parcel highlighting
* reordered layer options alphabetically
* added link to Maricopa County Recorders office for MCR docs

## Version 3.7.4 | 02/08/2016

* updated PDF reports URL for employment
* moved reports URL's from view to config
* updated grunt files

## Version 3.7.3 | 01/08/2016

* changed the title on the reports window to `Wickenburg`

## Version 3.7.2 | 01/08/2015

* Changed multiple parcel selection feature
* updated Employers to 2014 MAG employees from MAG employment database, +5 employees

## Version 3.7.1 (10/27/2015)

* fixed popup link so it shows only on parcels popup
* added MPA boundary to map, layers and TOC
* added ability to change parcel border thickness.
* added ability to highlight multiple parcels.
* moved zoning definitions from main navbar to contents panel.
* added Flood Zone Definitions link to contents window when floodzone layer is chosen.
* added reports button next to print button that links to wickenburg demographic and employment reporting.
* upgrade modernizr to `v2.8.3` via CDN - [modernizr](http://cdnjs.com/libraries/modernizr)
* upgrade html5shiv to `v3.7.3` via CDN - [html5shiv](https://cdnjs.com/libraries/html5shiv)
* upgrade respond.js to `v1.4.2` via CDN - [respond.js](https://cdnjs.com/libraries/respond.js)

## Version 3.7.0 (08/03/2015)

* added block group layer
* added popup for block group layer

## Version 3.6.0 (07/29/2015)

* add pending floodplain data
* fix flood zone TOC label
* add popup for pending floodplain data

## Version 3.5.1 (03/20/2015)

* update of parcel layer
* update of Zoning Layer

## Version 3.5.0 (03/05/2015)

* added code to hide popup when info not found
* made changes to definitions table
* made changes to popup style
* updated MXD
* updates to data - Zoning Info

## Version 3.4.0 (07/22/2014)

* changed size of search box (reduced)
* changed lods to minZoom & maxZoom
* fixed layer options to only display popup when layer is visible

## Version 3.3.0 (06/30/2014)

* changed size and location of basemap toggle
* added version number and date to about.html - configurable in appConfig file
* added print button
* added print functionality
* changed the style of panels and modals to dark
* added source reference to employer data
* made print legend static graphic to fit to size

## Version 3.2.1 (06/04/2014)

* added Wickenburg Employers, 5+ employees as a feature layer
* added Google Analytics code to index files `UA-40006601-1`
* added Google Webmaster Tools `google14c4dad344d14f91.html` file to root directory

## Version 3.2.0 (05/27/2014)

* moved `About` modal from index.html to views `about.html`
* moved `Measurement` html from index.html to views `measureTool.html`
* moved `Content` html from index.html to views `contents.html`

## Version 3.1.0 (05/12/2014)

* made changes to boundary and zoning

## Version 3.0.0 (05/09/2014)

* updating zoning - new annexation area
* updating boundary - new annexation area
* reordered zoning layer
* use `<!doctype html>` instead of `<!DOCTYPE html>`
* remove IE conditional classes per HTML5 Boilerplate
* update to [HTML5 Boilerplate v4.3.0](https://github.com/h5bp/html5-boilerplate) features
* update to [Normalize.css v3.0.1](https://github.com/h5bp/html5-boilerplate) per HTML5bp
* update the [HTML5 Shiv](https://github.com/aFarkas/html5shiv)
* update the [respond.min.js](https://github.com/scottjehl/Respond)

Gruntfile.js
module.exports = function (grunt) {

    "use strict";

    require("matchdep").filterDev("grunt-*").forEach(grunt.loadNpmTasks);

    grunt.initConfig({

        pkg: grunt.file.readJSON("package.json"),

        bannercss: "/*! ========================================================================\n" +
            " * Maricopa Association of Governments\n" +
            " * CSS files for MAG Wickenburg Zoning Map Viewer\n" +
            " * @concat.min.css | @version | <%= pkg.version %>\n" +
            " * Production | <%= pkg.date %>\n" +
            " * MAG Wickenburg Zoning Map Viewer\n" +
            " * http://geo.azmag.gov/maps/wickenburg/\n" +
            " * ==========================================================================\n" +
            " * @Copyright <%= pkg.copyright %> MAG\n" +
            " * @License MIT\n" +
            " * ========================================================================== */\n",

        bannerjs: "/*! ========================================================================\n" +
            " * Maricopa Association of Governments\n" +
            " * JavaScript files for MAG Wickenburg Zoning Map Viewer\n" +
            " * @main.min.js | @version | <%= pkg.version %>\n" +
            " * Production | <%= pkg.date %>\n" +
            " * MAG Wickenburg Zoning Map Viewer\n" +
            " * http://geo.azmag.gov/maps/wickenburg/\n" +
            " * ==========================================================================\n" +
            " * @Copyright <%= pkg.copyright %> MAG\n" +
            " * @License MIT\n" +
            " * ========================================================================== */\n",

        jshint: {
            files: ["src/js/config.js", "src/js/main.js"],
            options: {
                // strict: true,
                sub: true,
                quotmark: "double",
                trailing: true,
                curly: true,
                eqeqeq: true,
                unused: true,
                scripturl: true, // This option defines globals exposed by the Dojo Toolkit.
                dojo: true, // This option defines globals exposed by the jQuery JavaScript library.
                jquery: true, // Set force to true to report JSHint errors but not fail the task.
                force: true,
                reporter: require("jshint-stylish-ex")
            }
        },

        uglify: {
            options: {
                // add banner to top of output file
                banner: "<%= bannerjs %>\n"
            },
            build: {
                files: {
                    "dist/js/main.min.js": ["dist/js/main.js"],
                    "dist/js/config.min.js": ["dist/js/config.js"],
                    "dist/js/vendor/bootstrapmap.min.js": ["dist/js/vendor/bootstrapmap.js"],
                    "dist/js/vendor/plugins.min.js": ["dist/js/vendor/plugins.js"]
                }
            }
        },

        cssmin: {
            add_banner: {
                options: {
                    // add banner to top of output file
                    banner: '/* <%= pkg.name %> - v<%= pkg.version %> | <%= grunt.template.today("mm-dd-yyyy") %> */\n'
                },
                files: {
                    "dist/css/main.min.css": ["dist/css/main.css"],
                    "dist/css/normalize.min.css": ["dist/css/normalize.css"],
                    "dist/css/bootstrapmap.min.css": ["dist/css/bootstrapmap.css"]
                }
            }
        },

        concat: {
            options: {
                stripBanners: true,
                banner: "<%= bannercss %>\n"
            },
            dist: {
                src: ["dist/css/normalize.min.css", "dist/css/bootstrapmap.min.css", "dist/css/main.min.css"],
                dest: "dist/css/concat.min.css"
            }
        },

        clean: {
            build: {
                src: ["dist/"]
            },
            js: ["dist/js/*.js", "!dist/js/*.min.js"],
            jsv: ["dist/js/vendor/*.js", "!dist/js/vendor/*.min.js"],
            css: ["dist/css/*.css", "!dist/css/concat.min.css"]
        },

        copy: {
            build: {
                cwd: "src/",
                src: ["**"],
                dest: "dist/",
                expand: true
            }
        },

        watch: {
            scripts: {
                files: ["src/js/main.js", "src/js/config.js", "Gruntfile.js"],
                tasks: ["jshint"],
                options: {
                    spawn: false,
                    interrupt: true,
                },
            },
        },

        versioncheck: {
            options: {
                skip: ["semver", "npm", "lodash"],
                hideUpToDate: false
            }
        },

        toggleComments: {
            customOptions: {
                options: {
                    removeCommands: true
                },
                files: {
                    "dist/index.html": "src/index.html",
                    "dist/js/main.js": "src/js/main.js"

                }
            }
        },

        replace: {
            update_Meta: {
                src: ["src/index.html", "src/js/config.js", "src/humans.txt", "README.md", "src/LICENSE", "LICENSE"],
                overwrite: true, // overwrite matched source files
                replacements: [{
                    // html pages
                    from: /(<meta name="revision-date" content=")[0-9]{4}-[0-9]{2}-[0-9]{2}(">)/g,
                    to: '<meta name="revision-date" content="' + '<%= pkg.date %>' + '">',
                }, {
                    // html pages
                    from: /(<meta name="version" content=")([0-9]+)(?:\.([0-9]+))(?:\.([0-9]+))(">)/g,
                    to: '<meta name="version" content="' + '<%= pkg.version %>' + '">',
                }, {
                    // config.js
                    from: /(v)([0-9]+)(?:\.([0-9]+))(?:\.([0-9]+))( \| )[0-9]{4}-[0-9]{2}-[0-9]{2}/g,
                    to: 'v' + '<%= pkg.version %>' + ' | ' + '<%= pkg.date %>',
                }, {
                    // humans.txt
                    from: /(Version\: )([0-9]+)(?:\.([0-9]+))(?:\.([0-9]+))/g,
                    to: "Version: " + '<%= pkg.version %>',
                }, {
                    // humans.txt
                    from: /(Last updated\: )[0-9]{4}-[0-9]{2}-[0-9]{2}/g,
                    to: "Last updated: " + '<%= pkg.date %>',
                }, {
                    // LICENSE
                    from: /(Copyright \(c\) )[0-9]{4}/g,
                    to: "Copyright (c) " + "<%= pkg.copyright %>",
                }, {
                    // README.md
                    from: /(### version )([0-9]+)(?:\.([0-9]+))(?:\.([0-9]+))/g,
                    to: "### version " + '<%= pkg.version %>',
                }, {
                    // README.md
                    from: /(`Updated: )[0-9]{4}-[0-9]{2}-[0-9]{2}/g,
                    to: "`Updated: " + '<%= pkg.date %>',
                }, {
                    // main.css
                    from: /(@main.css)( \| )(version)( \| )([0-9]+)(?:\.([0-9]+))(?:\.([0-9]+))/g,
                    to: "@main.css | version |" + ' <%= pkg.version %>',
                }]
            }
        }


    });
    //==============================================================================>
    // this would be run by typing "grunt test" on the command line

    grunt.registerTask("check", ["versioncheck"]);
    grunt.registerTask("work", ["jshint"]);
    grunt.registerTask("workcss", ["csslint"]);

    grunt.registerTask("buildcss", ["cssmin", "concat"]);
    grunt.registerTask("buildjs", ["uglify"]);

    grunt.registerTask("tg", ["toggleComments"]);

     grunt.registerTask("build", ["clean:build", "replace", "copy", "toggleComments", "uglify", "cssmin", "concat", "clean:js", "clean:jsv", "clean:css"]);


    // the default task can be run just by typing "grunt" on the command line
    grunt.registerTask("default", []);

};

// ref
//==============================================================================>
// http://coding.smashingmagazine.com/2013/10/29/get-up-running-grunt/
// http://csslint.net/about.html
// http://www.jshint.com/docs/options/


app/vms/geoCode-vm.js
/**
 * contains geocode information
 *
 * @class geocode vm
 */

(function () {

    "use strict";

    define([
        "dojo/dom-construct",
        "dojo/topic",
        "dojo/dom",
        "dojo/on",
        "esri/dijit/Geocoder",
        "app/model/main.js",
        "esri/graphic",
        "esri/geometry/Multipoint",
        "esri/symbols/PictureMarkerSymbol",
        "esri/InfoTemplate",
        "esri/dijit/Popup",
        "dojo/parser",
        "dojo/domReady!"

    ],
        function (dc, tp, dom, on, Geocoder, mapModel, Graphic, Multipoint, PictureMarkerSymbol, InfoTemplate, Popup, parser) {

            var geoCodeVM = new function () {

                /**
                 * Store reference to module this object.
                 *
                 * @property self
                 * @type {*}
                 */
                var self = this;


                /**
                 * Initialize the class.
                 *
                 * @method init
                 */
                self.init = function () {

                    // create geosearch widget
                    var geocoder = new Geocoder({
                        value: '',
                        maxLocations: 10,
                        autoComplete: true,
                        arcgisGeocoder: true,
                        map: mapModel.map
                    }, "geosearch");
                    geocoder.startup();
                    geocoder.on("select", geocodeSelect);
                    geocoder.on("findResults", geocodeResults);
                    geocoder.on("clear", clearFindGraphics);

                } // end init
//****************************************************************

                self.geosearch = function () {
                    var def = geocoder.find();
                    def.then(function(res) {
                        geocodeResults(res);
                    });
                }

                self.geocodeSelect = function (item) {
                    var g = (item.graphic ? item.graphic : item.result.feature);
                    g.setSymbol(sym);
                    addPlaceGraphic(item.result, g.symbol);
                }

                self.geocodeResults = function (places){
                    places = places.results;
                    if (places.length > 0) {
                        clearFindGraphics();
                        var symbol = sym;
                        // Create and add graphics with pop-ups
                        for (var i = 0; i < places.length; i++) {
                            addPlaceGraphic(places[i], symbol);
                        }
                        zoomToPlaces(places);
                    } else {
                        alert("Sorry, address or place not found.");
                    }
                }

                self.addPlacesGraphic = function (item, symbol) {
                    var place = {};
                    var attributes, infoTemplate, pt, graphic;
                    pt = item.feature.geometry;
                    place.address = item.name;
                    place.score = item.feature.attributes.Score;
                    // Graphic components
                    attributes = {
                        address: place.address,
                        score: place.score,
                        lat: pt.getLatitude().toFixed(2),
                        lon: pt.getLongitude().toFixed(2)
                    };
                    infoTemplate = new InfoTemplate("${address}", "Latitude: ${lat}<br/>Longitude: ${lon}<br/>Score: ${score}");
                    graphic = new Graphic(pt, symbol, attributes, infoTemplate);
                    // Add to map
                    mapModel.map.graphics.add(graphic);
                }

                self.zoomToPlaces = function (places) {
                    var multiPoint = new Multipoint(mapModel.map.spatialReference);
                    for (var i = 0; i < places.length; i++) {
                        //multiPoint.addPoint(places[i].location);
                        multiPoint.addPoint(places[i].feature.geometry);
                    }
                    mapModel.map.setExtent(multiPoint.getExtent().expand(2.0));
                }

                self.clearFindGraphics = function () {
                    mapModel.map.infoWindow.hide();
                    mapModel.map.graphics.clear();
                }

                self.createPictureSymbol = function (url, xOffset, yOffset) {
                    return new PictureMarkerSymbol({
                        "angle": 0,
                        "xoffset": xOffset,
                        "yoffset": yOffset,
                        "type": "esriPMS",
                        "url": url,
                        "contentType": "image/png",
                        "width": 12,
                        "height": 24
                    });
                }
                var sym = self.createPictureSymbol("img/blue-pin.png", 0, 12, 35);


            }; // end geoCodeVM

            return geoCodeVM;

      } // end function
    )
} ());

src/js/config.js
/* ========================================================================
 * Maricopa Association of Governments
 * JavaScript file for MAG Wickenburg Zoning Map Viewer
 * @config.js
 * http://geo.azmag.gov/maps/wickenburg/
 * MAG Wickenburg Zoning Map Viewer
 * ==========================================================================
 * @Copyright 2017 MAG
 * @License MIT
 * ========================================================================== */

var appConfig = new function() {

    this.mainURL = "https://geo.azmag.gov/arcgis/rest/services/maps/WickenburgData/MapServer";

    this.layerInfo = [{
        layerNum: 0,
        id: "wiZoning",
        url: this.mainURL + "/8",
        title: "Zoning",
        visible: false,
        opacity: 0.6,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "Wickenburg Zoning",
        popupBody: "Zoning Code: ${CODE}" + "<br>Zoning Description: ${DESCRIPTION}",
        tocOrder: 1,
        legendOrder: 1,
        mapOrder: 11
    }, {
        layerNum: 1,
        id: "wiFlood",
        url: this.mainURL + "/5",
        title: "Flood Zone",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "Flood Zone",
        popupBody: "Flood Zone: ${ZONE}",
        tocOrder: 7,
        legendOrder: 6,
        mapOrder: 6
    }, {
        layerNum: 2,
        id: "wiPendFlood",
        url: this.mainURL + "/6",
        title: "Flood Zone Pending",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "Flood Zone Pending",
        popupBody: "Pending Flood Zone: ${FloodZone}",
        tocOrder: 6,
        legendOrder: 7,
        mapOrder: 5
    }, {
        layerNum: 3,
        id: "tParcels",
        url: this.mainURL + "/2",
        title: "Parcels",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "Parcels",
        popupBody: "County: ${COUNTY}<br>" + "Parcel: ${PARCEL}<br>" + "Address: ${ADDRESS}",
        tocOrder: 4,
        legendOrder: 4,
        mapOrder: 4
    }, {
        layerNum: 4,
        id: "coBoundary",
        url: this.mainURL + "/4",
        title: "County Boundary",
        visible: true,
        opacity: 1,
        showLegend: true,
        showCheckBox: false,
        popupHeader: "",
        popupBody: "",
        legendOrder: 11,
        mapOrder: 10
    }, {
        layerNum: 5,
        id: "wiBoundary",
        url: this.mainURL + "/3",
        title: "Town Boundary",
        visible: true,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "",
        popupBody: "",
        tocOrder: 2,
        legendOrder: 8,
        mapOrder: 8
    }, {
        layerNum: 6,
        id: "wiMPA",
        url: this.mainURL + "/10",
        title: "Municipal Planning Area",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "",
        popupBody: "",
        tocOrder: 5,
        legendOrder: 9,
        mapOrder: 9
    }, {
        layerNum: 7,
        id: "wiBlockGroups",
        url: this.mainURL + "/9",
        title: "Block Groups",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "Wickenburg Block Groups",
        popupBody: "<strong>GEOID:  ${GEOID10}</strong><br>" + "<b>Total Persons:</b>  ${LOWMODUNIV:NumberFormat}<br>" + "<b>Low Income:</b>  ${LOW:NumberFormat}<br>" + "<b>Low & Moderate Income:</b>  ${LOWMOD:NumberFormat}<br>" + "<b>Low, Moderate, & Medium Income:</b>  ${LMMI:NumberFormat}<br>" + "<b>% Low & Moderate Income:</b> ${LOWMOD_PCT:NumberFormat}%",
        tocOrder: 10,
        legendOrder: 10,
        mapOrder: 7
    }, {
        layerNum: 8,
        id: "wiEmployers",
        url: this.mainURL + "/1",
        title: "Employers, 5+ employees",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "Employers",
        popupBody: "<strong>${EmpName}</strong><hr class='pLine'>${Address}</br>" + "${Jurisdiction}, ${State} ${Zip}<br>" + "Type:  ${Cluster}",
        tocOrder: 8,
        legendOrder: 3,
        mapOrder: 2
    }, {
        layerNum: 9,
        id: "wiBLI",
        url: this.mainURL + "/0",
        title: "Building & Landmark Inventory",
        visible: true,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "${Name}",
        popupBody: "<hr class='pLine'> Category:  ${BLI_Cat}<br> Sub Category: ${SubCat}<br> Website: ${Website}",
        tocOrder: 9,
        legendOrder: 2,
        mapOrder: 1
    }, {
        layerNum: 10,
        id: "wiSubdivisions",
        url: this.mainURL + "/7",
        title: "Subdivisions",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "${NAME}",
        popupBody: "<hr class='pLine'> MCR:  ${MCR}",
        tocOrder: 3,
        legendOrder: 5,
        mapOrder: 3
    }, {
        layerNum: 11,
        id: "wiOpportunityZones",
        url: this.mainURL + "/11",
        title: "Opportunity Zones",
        visible: false,
        opacity: 1,
        showLegend: true,
        showCheckBox: true,
        popupHeader: "${NAMELSAD10}",
        popupBody: "<hr class='pLine'> Name:  ${NAME10}<br> SQMI: ${SQMI}",
        tocOrder: 0,
        legendOrder: 12,
        mapOrder: 12
    }];

    this.streetsLayer = "https://server.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer";
    this.imageryLayer = "https://server.arcgisonline.com/arcgis/rest/services/World_Imagery/MapServer";

    this.Version = "v3.9.1 | 2020-02-14";

    this.copyright = "2019";

    this.emailLink = "https://www.azmag.gov/EmailPages/JasonHoward.asp";

    this.MaricopaAssessor = "https://mcassessor.maricopa.gov/mcs.php?q=";
    this.YavapaiAssessor = "https://gis.yavapai.us/v4/print_parcel.aspx?qs=";

    this.MaricopaRecorder = "https://recorder.maricopa.gov/recdocdata/GetBookMap.aspx?docket=";

    this.GeometryService = "https://tasks.arcgisonline.com/ArcGIS/rest/services/Geometry/GeometryServer"; // Esri test service
    // this.GeometryService = "https://geo.azmag.gov/gismag/rest/services/Utilities/Geometry/GeometryServer"; // MAGs service

    this.LocatorService = "//geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer";

    // this.printUrl = "https://geo.azmag.gov/gismag/rest/services/gp/Wickenburg_Print/GPServer/Export%20Web%20Map"; // MAG Wickenburg Print service
    this.printUrl = "http://sampleserver6.arcgisonline.com/arcgis/rest/services/Utilities/PrintingTools/GPServer/Export%20Web%20Map%20Task"; // Esri test service

    // Demographic PDF report
    this.demService = "https://geo.azmag.gov/services/demographics2015/reports.html?city=Wickenburg";
    // Employment PDF report
    this.empService = "https://geo.azmag.gov/services/employment2016/reports.html?jurisdiction=Wickenburg";

    this.initExtent = {
        "xmin": -12570767,
        "ymin": 4018810,
        "xmax": -12534077,
        "ymax": 4037517,
        "spatialReference": {
            "wkid": 102100
        }
    };

    this.center = [-112.343, 33.285]; // center on Wickenburg

    // ------------------------------
    // MARKUP TOOL SETTINGS
    // ------------------------------

    // Specify the Markup / drawing tools.
    this.markupToolTreeNodes =
        [{
        id: 1,
        text: "Polygon",
        DisplayText: "Polygon",
        Type: "POLYGON"
    }, {
        id: 2,
        text: "Circle",
        DisplayText: "Circle",
        Type: "CIRCLE"
    }, {
        id: 3,
        text: "Arrow",
        DisplayText: "Arrow",
        Type: "ARROW"
    }, {
        id: 4,
        text: "Freehand",
        DisplayText: "Freehand Polygon",
        Type: "FREEHAND_POLYGON"
    }, {
        id: 5,
        text: "Text",
        DisplayText: "Text",
        Type: "POINT"
    }];

    this.fillColorOpacity = 0.75;

    this.textSymbolFontSizes = [8, 9, 10, 11, 12, 14, 16, 18, 20, 24, 30, 50];

    this.defaultOutlineColor = [0, 0, 0];

    this.defaultFillColor = [0, 0, 255];

}; //End Config
